plainly-plugin/src/ui/components/form/UploadForm.tsx

```diff
@@ -75,6 +104,56 @@ export default function UploadForm() {
     }
   };
 
+  useEffect(() => {
+    const fetchProject = async (
+      key: string,
+      projectId: string,
+      revision: string,
+    ) => {
+      try {
+        const { data } = await get<Project>(
+          `/api/v2/projects/${projectId}`,
+          key,
+        );
+
+        const revisionHistory = data.revisionHistory;
+        const latestRevision = revisionHistory?.length || 0;
+
+        if (latestRevision.toString() !== revision) {
+          setBadRevision(true);
+        }
+
+        setProjectExists(true);
+        notifySuccess('Project fetched successfully');
+      } catch (error) {
+        setProjectExists(false);
+        notifyError('Failed to fetch project', (error as Error).message);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    if (uploadMode === 'edit' && projectData?.id && projectData.revision) {
+      setLoading(true);
+      fetchProject(apiKey, projectData.id, projectData.revision);
```

Comment:
* imo fetch project should just fetch project and can be idenpendent useCAllable in this component:
* should not know anything else about revisions and handling states etc
* plus you should save this project to the state and then you can extract all the needed info like if it's bad revision, you can fill data (name description) in edit mode, etc
```suggestion
      try {
          const .. = awaitt fetchProject(apiKey, projectData.id, projectData.revision);
          // handle fetch result
      }     
```

Reply:
same as below, other PR where we will handle this with react query? This one is getting big, lets just fix commnets above, and make it work, and then in seperate PR clean this code, and all the other fetches around the app

Reply:
moved to tanstack query